modules/core/include/opencv2/core/persistence.hpp

```diff
@@ -767,6 +767,17 @@ template<typename _Tp, int cn> static inline void read(const FileNode& node, Vec
     value = temp.size() != cn ? default_value : Vec<_Tp, cn>(&temp[0]);
 }
 
+template<typename _Tp, int m, int n> static inline void read(const FileNode& node, Matx<_Tp, m, n>& value, const Matx<_Tp, m, n>& default_matx = Matx<_Tp, m, n>())
+{
+    Mat temp;
+    read(node, temp); // read as a Mat class
+
+    if ( !temp.data || temp.dims > 2 || temp.rows != m || temp.cols != n || temp.channels() != 1 )
+        value = default_matx;
```

Comment:
1. To catch errors early we should throw an error here.
Only empty Mat (missing file node) can replaced with default value.
Perhaps `Matx<_Tp, m, n>()` ctor is able to raise most part of errors, so condition should looks like:
```
if (temp.empty())
    value = default_matx;
else
    value = Matx<_Tp, m, n>(temp);
```

2. Missing type check in this condition. Not sure that we really want to cast bytes (8U) into floats implicitly.

Reply:
1. That's fine by me. I based the condition on the Vec read function (just above) which also returns default if the size does not match, but happy to have the Mat constructor throw an error in this condition if preferred.

2. I thought this would be handled sensibly by the Matx constructor, but I can't find the code that would run when a Mat is passed? Any suggestions welcome, even if just reject if Mat and Matx types are not the same (though this would seem a little harsh, i.e. float --> double).